server/datastore/mysql/policies.go

```diff
@@ -289,112 +293,76 @@ func (ds *Datastore) RecordPolicyQueryExecutions(ctx context.Context, host *flee
 }
 
 func (ds *Datastore) ListGlobalPolicies(ctx context.Context, opts fleet.ListOptions) ([]*fleet.Policy, error) {
-	return listPoliciesDB(ctx, ds.reader(ctx), nil, nil, opts)
+	return listPoliciesDB(ctx, ds.reader(ctx), nil, opts)
 }
 
 // returns the list of policies associated with the provided teamID, or the
-// global policies if teamID is nil. The pass/fail host counts are the totals
-// regardless of hosts' team if countsForTeamID is nil, or the totals just for
-// hosts that belong to the provided countsForTeamID if it is not nil.
-func listPoliciesDB(ctx context.Context, q sqlx.QueryerContext, teamID, countsForTeamID *uint, opts fleet.ListOptions) ([]*fleet.Policy, error) {
+// global policies if teamID is nil
+func listPoliciesDB(ctx context.Context, q sqlx.QueryerContext, teamID *uint, opts fleet.ListOptions) ([]*fleet.Policy, error) {
```

Comment:
A lot of simplification work done here.  No longer need to calculate counts, and I also moved inherited policy queries to a new function (below)